client/src/App.js

```diff
@@ -22,9 +20,7 @@ function App() {
           <Route path="/signup" component={SignUp} />
           <Route path="/calendar" component={CalendarPage} />
           <Route exact path={["/", "/home", "/home/:page"]} component={Home} />
-          <Route exact path="/profile-settings" component={ProfileSetting} />
-          <Route exact path="/calendar-confirm" component={CalendarConfirm} />
-          <Route exact path="/availability" component={UserAvailability} />
+          <Route path="/onboarding" component={Onboarding} />
```

Comment:
@sina-jamshidi @bugatti12345 I am not sure what to do with path of the parent component. Is there a way for me to hide this or redirect it to the first step of onboarding whenever user tries to access /onboarding?

Reply:
I think we only need the` /onboarding` path there.

When user tries to access onboarding:
1. Get which user active step it should be from backend (ex. if url for that user is `null` in DB, the active step should be 1).
2. Set the active step state before render
3. Based on the active step, render which child component it should be. (ex. `activestep===1 => render ProfileSetting`)

Since users can't skip steps and they have to follow the active step and pick it up where they haven't done.

That's my logic, don't know if it helps.

Reply:
yep thats definitely another way to do it, but I was trying to see if there is a way to use React routes so that users can toggle steps using the browser's back and forth buttons.

Reply:
Oh I see. So maybe you can set like `<Route exact path={["/profile-settings", "/calendar-confirm", "/availability"]} component={(props)=><Onboarding location={props.location} />} />` in `app.js` to pass the path to onboarding, and get the `path` in onboarding by `props.location.pathname`

Reply:
We can always have nested routes too. So there can be another router inside Onboarding that displays the appropriate component based on the route.
Otherwise checking props.location works too

Reply:
tyty